Rename FrameLoadType::Replace to FrameLoadType::MultipartReplace

RupinMittal opened the pull request:
#### 1dc197ee54c4681c607ef9d6e7f89f62ee5e2d50
<pre>
Rename FrameLoadType::Replace to FrameLoadType::MultipartReplace
<a href="https://bugs.webkit.org/show_bug.cgi?id=299497">https://bugs.webkit.org/show_bug.cgi?id=299497</a>
<a href="https://rdar.apple.com/161284247">rdar://161284247</a>

Reviewed by Brady Eidson.

Rename FrameLoadType::Replace to FrameLoadType::MultipartReplace to make it
clearer what this type is for. We also rename HistoryController::updateForReload
to HistoryController::updateForReloadOrReplace for clarity.

This is in preparation for introducting a new FrameLoadType for the Navigation
API Replace operation in a subsequent patch. We want to distinguish this from
MultipartReplace.

No new tests as this is just a rename. There should be no behavior change.

* Source/WebCore/history/BackForwardCache.cpp:
(WebCore::canCachePage):
* Source/WebCore/loader/DocumentLoader.cpp:
(WebCore::DocumentLoader::responseReceived):
(WebCore::DocumentLoader::commitData):
(WebCore::DocumentLoader::setupForMultipartReplace):
(WebCore::DocumentLoader::isMultipartReplacingLoad const):
(WebCore::DocumentLoader::maybeFinishLoadingMultipartContent):
(WebCore::DocumentLoader::setupForReplace): Deleted.
* Source/WebCore/loader/DocumentLoader.h:
* Source/WebCore/loader/FrameLoader.cpp:
(WebCore::isBackForwardLoadType):
(WebCore::isReload):
(WebCore::determineNavigationType):
(WebCore::FrameLoader::setupForMultipartReplace):
(WebCore::FrameLoader::commitProvisionalLoad):
(WebCore::FrameLoader::transitionToCommitted):
(WebCore::FrameLoader::isMultipartReplacing const):
(WebCore::FrameLoader::setMultipartReplacing):
(WebCore::FrameLoader::subresourceCachePolicy const):
(WebCore::FrameLoader::checkLoadCompleteForThisFrame):
(WebCore::FrameLoader::loadDifferentDocumentItem):
(WebCore::FrameLoader::setupForReplace): Deleted.
(WebCore::FrameLoader::isReplacing const): Deleted.
(WebCore::FrameLoader::setReplacing): Deleted.
* Source/WebCore/loader/FrameLoader.h:
* Source/WebCore/loader/FrameLoaderTypes.h:
* Source/WebCore/loader/HistoryController.cpp:
(WebCore::HistoryController::restoreDocumentState):
(WebCore::HistoryController::updateForReloadOrReplace):
(WebCore::HistoryController::updateForCommit):
(WebCore::HistoryController::isMultipartReplaceLoadTypeWithProvisionalItem):
(WebCore::HistoryController::updateForReload): Deleted.
(WebCore::HistoryController::isReplaceLoadTypeWithProvisionalItem): Deleted.
* Source/WebCore/loader/HistoryController.h:
* Source/WebCore/page/Page.cpp:
(WebCore::Page::logNavigation):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
(toWebFrameLoadType):

Canonical link: <a href="https://commits.webkit.org/300517@main">https://commits.webkit.org/300517@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/247fcc357707edcec8f792df68f95934a2c3d93d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/122809 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/131/builds/42619 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/33211 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/129435 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/74913 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/fa6c31c7-d89e-4100-b4c0-c1e8f079adb2) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/124685 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/43240 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/51114 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/93337 "7 flakes 16 failures") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/61962 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/e4956bb3-b1b8-43cb-92d7-8a2307d264ff) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/125761 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/34474 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/109933 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/73975 "Found 1 new API test failure: WPE/TestWebsiteData:/webkit/WebKitWebsiteData/itp (failure)") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/3be0741b-5f1c-43aa-af8f-68ed041a2190) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/33454 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/28090 "Passed tests") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/73067 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/104173 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/28303 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/132161 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/49754 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/37877 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/101842 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/50131 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/106149 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/101717 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/47100 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/25276 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/46525 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/19389 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/49612 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/55364 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/49078 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/52430 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/50761 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

beidson commented on Source/WebCore/loader/HistoryController.cpp:
```diff
@@ -519,7 +519,7 @@ void HistoryController::updateForBackForwardNavigation()
-void HistoryController::updateForReload()
+void HistoryController::updateForReloadOrReplace()
 {
     LOG(History, "HistoryController %p updateForReload: Updating History for reload in frame %p (main frame %d) %s", this, m_frame.ptr(), m_frame->isMainFrame(), m_frame->loader().documentLoader() ? m_frame->loader().documentLoader()->url().string().utf8().data() : "");
```
Update comment, too!

RupinMittal replied:
Nice catch! Fixed.

RupinMittal committed:
Rename FrameLoadType::Replace to FrameLoadType::MultipartReplace
https://bugs.webkit.org/show_bug.cgi?id=299497
rdar://161284247

Reviewed by Brady Eidson.

Rename FrameLoadType::Replace to FrameLoadType::MultipartReplace to make it
clearer what this type is for. We also rename HistoryController::updateForReload
to HistoryController::updateForReloadOrReplace for clarity.

This is in preparation for introducting a new FrameLoadType for the Navigation
API Replace operation in a subsequent patch. We want to distinguish this from
MultipartReplace.

No new tests as this is just a rename. There should be no behavior change.

* Source/WebCore/history/BackForwardCache.cpp:
(WebCore::canCachePage):
* Source/WebCore/loader/DocumentLoader.cpp:
(WebCore::DocumentLoader::responseReceived):
(WebCore::DocumentLoader::commitData):
(WebCore::DocumentLoader::setupForMultipartReplace):
(WebCore::DocumentLoader::isMultipartReplacingLoad const):
(WebCore::DocumentLoader::maybeFinishLoadingMultipartContent):
(WebCore::DocumentLoader::setupForReplace): Deleted.
* Source/WebCore/loader/DocumentLoader.h:
* Source/WebCore/loader/FrameLoader.cpp:
(WebCore::isBackForwardLoadType):
(WebCore::isReload):
(WebCore::determineNavigationType):
(WebCore::FrameLoader::setupForMultipartReplace):
(WebCore::FrameLoader::commitProvisionalLoad):
(WebCore::FrameLoader::transitionToCommitted):
(WebCore::FrameLoader::isMultipartReplacing const):
(WebCore::FrameLoader::setMultipartReplacing):
(WebCore::FrameLoader::subresourceCachePolicy const):
(WebCore::FrameLoader::checkLoadCompleteForThisFrame):
(WebCore::FrameLoader::loadDifferentDocumentItem):
(WebCore::FrameLoader::setupForReplace): Deleted.
(WebCore::FrameLoader::isReplacing const): Deleted.
(WebCore::FrameLoader::setReplacing): Deleted.
* Source/WebCore/loader/FrameLoader.h:
* Source/WebCore/loader/FrameLoaderTypes.h:
* Source/WebCore/loader/HistoryController.cpp:
(WebCore::HistoryController::restoreDocumentState):
(WebCore::HistoryController::updateForReloadOrReplace):
(WebCore::HistoryController::updateForCommit):
(WebCore::HistoryController::isMultipartReplaceLoadTypeWithProvisionalItem):
(WebCore::HistoryController::updateForReload): Deleted.
(WebCore::HistoryController::isReplaceLoadTypeWithProvisionalItem): Deleted.
* Source/WebCore/loader/HistoryController.h:
* Source/WebCore/page/Page.cpp:
(WebCore::Page::logNavigation):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
(toWebFrameLoadType):

Canonical link: https://commits.webkit.org/300517@main
